harmony/http.py

```diff
@@ -249,28 +252,46 @@ def download(config, url: str, access_token: str, data, destination_file):
 
     response = None
     logger = build_logger(config)
-    logger.info('Downloading %s', url)
+    start_time = datetime.datetime.now()
```

Comment:
The timing & logging _might_ work well as a decorator that we place on this or other methods/functions. That may help readability so the timing & logging aren't intertwined with the behavior to download, which is already a little complicated.